app/scripts/directives/util.js

```diff
@@ -103,6 +103,36 @@ angular.module('openshiftConsole')
       }
     };
   })
+  .directive('copyToClipboardButton', function(NotificationsService){
+    return {
+      restrict: 'E',
+      replace: true,
+      scope: {
+        clipboardText: "@",
+        buttonText: "@"
+      },
+      template: '<button class="clipboard-button pull-right btn btn-default" data-clipboard-text="{{clipboardText}}">{{buttonText}}</button>',
+      link: function($scope, element){
+        var clipboard = new Clipboard( element.get(0) );
+        clipboard.on('success', function (e) {
+          $(e.trigger)
+            .attr('title', 'Copied!')
+            .tooltip('fixTitle')
+            .tooltip('show')
+            .attr('title', 'Copy to Clipboard')
+            .tooltip('fixTitle');
+          e.clearSelection();
+        });
+        clipboard.on('error', function () {
+          NotificationsService.addNotification({
```

Comment:
`onError` binding that the user of the directive decides how to respond might be better here.